web/src/OverviewTable.tsx

```diff
@@ -0,0 +1,482 @@
+import React, { useState } from "react"
+import { CellProps, Column, useSortBy, useTable } from "react-table"
+import TimeAgo from "react-timeago"
+import styled from "styled-components"
+import { buildAlerts, runtimeAlerts } from "./alerts"
+import { incr } from "./analytics"
+import { ReactComponent as CheckmarkSvg } from "./assets/svg/checkmark.svg"
+import { ReactComponent as CopySvg } from "./assets/svg/copy.svg"
+import { ReactComponent as LinkSvg } from "./assets/svg/link.svg"
+import { InstrumentedButton } from "./instrumentedComponents"
+import { displayURL } from "./links"
+import TableStarResourceButton from "./OverviewTableStarResourceButton"
+import OverviewTableStatus from "./OverviewTableStatus"
+import TriggerButton from "./OverviewTableTriggerButton"
+import TableTriggerModeToggle from "./OverviewTableTriggerModeToggle"
+import { useResourceNav } from "./ResourceNav"
+import { useStarredResources } from "./StarredResourcesContext"
+import { buildStatus, runtimeStatus } from "./status"
+import {
+  Color,
+  Font,
+  FontSize,
+  mixinResetButtonStyle,
+  SizeUnit,
+} from "./style-helpers"
+import { isZeroTime, timeDiff } from "./time"
+import { timeAgoFormatter } from "./timeFormatters"
+import { ResourceStatus, TargetType, TriggerMode } from "./types"
+
+type UIResource = Proto.v1alpha1UIResource
+type UIResourceStatus = Proto.v1alpha1UIResourceStatus
+type Build = Proto.v1alpha1UIBuildTerminated
+type UILink = Proto.v1alpha1UIResourceLink
+
+type OverviewTableProps = {
+  view: Proto.webviewView
+}
+
+type RowValues = {
+  lastDeployTime: string
+  trigger: OverviewTableTrigger
+  name: string
+  resourceTypeLabel: string
+  statusLine: OverviewTableStatus
+  podId: string
+  endpoints: UILink[]
+  triggerMode: TriggerMode
+}
+
+type OverviewTableTrigger = {
+  isBuilding: boolean
+  hasBuilt: boolean
+  hasPendingChanges: boolean
+  isQueued: boolean
+}
+
+type OverviewTableStatus = {
+  buildStatus: ResourceStatus
+  buildAlertCount: number
+  lastBuildDur: moment.Duration | null
+  runtimeStatus: ResourceStatus
+  runtimeAlertCount: number
+}
+
+const ResourceTable = styled.table`
+  margin-top: ${SizeUnit(0.5)};
+  margin-left: ${SizeUnit(1)};
+  margin-right: ${SizeUnit(1)};
+  border-collapse: collapse;
+`
+const ResourceTableHead = styled.thead`
+  background-color: ${Color.grayDarker};
+`
+const ResourceTableRow = styled.tr`
+  border-bottom: 1px solid ${Color.grayLighter};
+`
+const ResourceTableData = styled.td`
+  font-family: ${Font.monospace};
+  font-size: ${FontSize.smallest};
+  color: ${Color.gray6};
+  box-sizing: border-box;
+`
+const ResourceTableHeader = styled(ResourceTableData)`
+  color: ${Color.gray7};
+  font-size: ${FontSize.smallester};
```

Comment:
This is a small note (no pun intended 🤦🏻). The headers are a little small for me to read as a user, even on smaller screens. What would you think about bumping this up to `smallest` instead of `smallester`? You may not want to mess with style tweaks at this stage of feedback, so happy to revisit this later!